src/main/java/org/codeforamerica/formflowstarter/app/data/Submission.java

```diff
@@ -90,4 +87,16 @@ public static Submission mergeFormDataWithSubflowIterationData(Submission submis
     submission.setInputData(existingInputData);
     return submission;
   }
+
+  public static void removeIncompleteIterations(Submission submission, String subflowName) {
+    List<Map<String, Object>> toRemove = new ArrayList<>();
+    ArrayList<Map<String, Object>> subflow = (ArrayList<Map<String, Object>>) submission.getInputData()
+        .get(subflowName);
+    for (int i = 0; i < subflow.size() - 1; i++) {
+      if (subflow.get(i).get("iterationIsComplete").equals(false)) {
```

Comment:
I see where this is set in the controller, but I do wonder if there's ever a case where this might not be set yet?   Should this check for `subflow.get(i).get("iterationIsComplete") == null` as well?   Maybe it will always be set though.

Reply:
Thanks for taking a looking and for the feedback :)

We set iterationIsComplete on the POST for both the /{flow}/{screen}/new endpoint and the /{flow}/{screen}/{uuid} endpoint so it shouldn't ever be null in this case, as the only time it wouldn't be there is when you first GET a subflow page. 

That said, we're refactoring this solution and working on some improvements and will keep this in mind as we work on that refactor.